iOS/RCTOrientation/Orientation.m

```diff
@@ -138,11 +138,17 @@ - (void)lockToOrientation:(UIInterfaceOrientation) newOrientation usingMask:(UII
     UIInterfaceOrientation deviceOrientation = _lastDeviceOrientation;
     
     [Orientation setOrientation:mask];
-    UIDevice* currentDevice = [UIDevice currentDevice];
-    
-    [currentDevice setValue:@(UIInterfaceOrientationUnknown) forKey:orientation];
-    [currentDevice setValue:@(newOrientation) forKey:orientation];
-    
+
+    if (@available(iOS 16.0, *)) {
+        NSArray *array = [[[UIApplication sharedApplication] connectedScenes] allObjects];
+        UIWindowScene *scene = (UIWindowScene *)array[0];    UIWindowSceneGeometryPreferencesIOS *geometryPreferences = [[UIWindowSceneGeometryPreferencesIOS alloc] initWithInterfaceOrientations:mask];
+        [scene requestGeometryUpdateWithPreferences:geometryPreferences errorHandler:^(NSError * _Nonnull error) { }];
```

Comment:
Can we make `lockToOrientation` accept

```
resolve: (RCTPromiseResolveBlock) resolve
reject: (RCTPromiseRejectBlock)reject
```

parameters and then reject the promise with a related error message in this error handler? For example:
`reject(@"err", [error description], nil);`